features/timeseries/src/main/java/org/opennms/netmgt/timeseries/integration/MetaTagConfiguration.java

```diff
@@ -0,0 +1,178 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2020 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2020 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.netmgt.timeseries.integration;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Defines which additional meta tags should be exposed to the timeseries integration plugin. */
+public class MetaTagConfiguration {
+
+    final static String PREFIX = "org.opennms.timeseries.tin.metatags";
+
+    /** properties defined in opennms.properties */
+    public enum PropertyKey {
+        assets,
+        tags,
+        categories
+    }
+
+    public enum MetaTagKey {
+        nodeLabel,
+        location,
+        sysObjectID,
+        foreignSource,
+        foreignId,
+        nodeCriteria,
+        ipAddress, // for response time resources
+        service, // for response time resources
+        ifDescr, // for interface resources
+        ifAlias, // for interface resources
+        resourceLabel
+    }
+
+    // TODO: Patrick: do we want to merge MetaTagKey & AssetTagKey?
```

Comment:
I think we should move this to the Meta-Data DSL instead of defining it here.

We could add these to a new "asset" context and make them available in expressions like `${asset:operatingSystem}`.